Shell/LineEditor.cpp

```diff
@@ -354,15 +388,43 @@ String LineEditor::get_line(const String& prompt)
                 }
 
                 String token = is_empty_token ? String() : String(&m_buffer[token_start], m_cursor - (size_t)token_start);
+                Vector<String> suggestions;
 
                 if (is_first_token)
-                    tab_complete_first_token(token);
+                    suggestions = tab_complete_first_token(token);
                 else
-                    tab_complete_other_token(token);
+                    suggestions = tab_complete_other_token(token);
+
+                if (m_times_tab_pressed > 1 && !suggestions.is_empty()) {
+                    size_t longest_name = 0;
+
+                    for (auto& suggestion : suggestions)
+                        longest_name = max(longest_name, suggestion.length());
+
+                    size_t num_printed = 0;
+                    putchar('\n');
+                    for (auto& suggestion : suggestions) {
+                        int new_pos = num_printed + suggestion.length() + (longest_name + 2 - suggestion.length());
```

Comment:
So just `num_printed + longest_name + 2`? Or is there an intuition behind putting it like this?

`new_pos` and `num_printed` could be named clearer, maybe `column` and `next_column` (or something, I'm not quite sure what it's meant to be?)

Reply:
> So just num_printed + longest_name + 2? Or is there an intuition behind putting it like this?
Classic 2am error :^)